### PR TITLE
add docs for shareable option

### PR DESCRIPTION
**Pull request by alicefr**
Document `shareable` option with an example

### CODE REVIEW

**Review by aglitke — APPROVED** [review]
Really complete example and explanation.  Nice!

File: docs/virtual_machines/disks_and_volumes.md [hunk 1]
```diff
@@ -1257,3 +1257,150 @@ Example: force `writethrough` cache mode
         persistentVolumeClaim:
           claimName: disk-alpine
     status: {}
+
+### Disk sharing
+
+Shareable disks allows multiple VMs to share the same underlying storage. In order to use this feature, a special care is required because this could lead to data corruption and the loss of important data. Shareable disks demand either data synchronization at application level or the usage of clustered filesystems. These advanced configurations are out-of-scope of this documentation and use-cases specific.
```
Comment (kgoldbla) [inline docs/virtual_machines/disks_and_volumes.md:1263]: a special care -> special care

Comment (kgoldbla) [inline docs/virtual_machines/disks_and_volumes.md:1263]: a special care -> special care
Shareable disks demand either data synchronization at application level or the usage of clustered filesystems -> Shareable disks demand either data synchronization at the application level or the use of clustered filesystems.
..are out-of-scope of this documentation and use-cases specific -> are not within the scope of this documentation and  are use-case specific.

File: docs/virtual_machines/disks_and_volumes.md [hunk 2]
```diff
@@ -1257,3 +1257,150 @@ Example: force `writethrough` cache mode
...
+    requests:
+      storage: 1Gi
+  storageClassName: rook-ceph-block
+```
+```bash
+$ kubectl get pvc
+NAME        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+block-pvc   Bound    pvc-0a161bb2-57c7-4d97-be96-0a20ff0222e2   1Gi        RWO            rook-ceph-block   51s
+```
+Then, we can declare 2 VMs and setting the `shareable` option to true for the shared disk.
```
Comment (kgoldbla) [inline docs/virtual_machines/disks_and_volumes.md:1287]: setting -> set

File: docs/virtual_machines/disks_and_volumes.md [hunk 3]
```diff
@@ -1257,3 +1257,150 @@ Example: force `writethrough` cache mode
...
+          userData: |-
+            #cloud-config
+            password: fedora
+            chpasswd: { expire: False }
+        name: cloudinitdisk
+      - name: block-disk
+        persistentVolumeClaim:
+          claimName: block-pvc                                        
+```
+We can try to write and read a string in the 2 guest to test the sharing.
```
Comment (kgoldbla) [inline docs/virtual_machines/disks_and_volumes.md:1391]: We can try to write and read a string in the 2 guest to test the sharing -> We can now attempt to write a string from the first host and then read the string from the second host to test that the sharing is working.

File: docs/virtual_machines/disks_and_volumes.md [hunk 4]
```diff
@@ -1257,3 +1257,150 @@ Example: force `writethrough` cache mode
...
+28 bytes copied, 0.0264182 s, 1.1 kB/s
+# Log into the second guest
+$ virtctl console vm-block-2
+$ sudo dd  if=/dev/vdc bs=1 count=150 conv=notrunc
+Test awesome shareable disks150+0 records in
+150+0 records out
+150 bytes copied, 0.136753 s, 1.1 kB/s
+```
+
+If you are using local devices or RWO PVCs, putting the affinity on the VMs that share the storage guarantees they will be scheduled on the same node. In the example, we put the affinity on the second VM using the label set on the first VM. If you are using shared storage with RWX PVCs, then the affinity rule is not necessary as the storage can be attached simoultaneously on multiple nodes.
```
Comment (kgoldbla) [inline docs/virtual_machines/disks_and_volumes.md:1406]: putting -> setting
In the example, we put the affinity -> In the example, we set the affinity
simoultaneously - simultaneously

**Review by kgoldbla — CHANGES_REQUESTED** [review]

File: docs/virtual_machines/disks_and_volumes.md (hunk 4 above)
Comment (jobbler) [reply docs/virtual_machines/disks_and_volumes.md:1406]: also change "putting the affinity on the VMs" -> "setting the affinity on the VMs"

**Comment by alicefr** [comment]
Update the docs with the correction

File: docs/virtual_machines/disks_and_volumes.md [hunk 5]
```diff
@@ -1257,3 +1257,150 @@ Example: force `writethrough` cache mode
         persistentVolumeClaim:
           claimName: disk-alpine
     status: {}
+
+### Disk sharing
+
+Shareable disks allows multiple VMs to share the same underlying storage. In order to use this feature, special care is required because this could lead to data corruption and the loss of important data. Shareable disks demand either data synchronization at the application level or the use of clustered filesystems. These advanced configurations are not within the scope of this documentation and are use-case specific..
```
Comment (kgoldbla) [inline docs/virtual_machines/disks_and_volumes.md:1263]: Shareable disks allows -> Shareable disks allow
specific**..** -> specific. (remove second full stop)

**Review by kgoldbla — CHANGES_REQUESTED** [review]
Two minor changes, the rest looks fine.

**Comment by alicefr** [comment]
@kgoldbla many thanks. I addressed the last comment, could you please have another look?

**Review by kgoldbla — APPROVED** [review]
/lgtm

**Review by aglitke — APPROVED** [review]
/lgtm

**Comment by aglitke** [comment]
@davidvossel this PR is ready for merge.  I'd appreciate you bumping it along.  Thanks.

**Pull request merged**